core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java

```diff
@@ -0,0 +1,96 @@
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.opentosca.toscana.core.testdata.TestCsars.CSAR_YAML_VALID_DOCKER_SIMPLETASK;
+
+public class KubernetesPluginTest extends BaseSpringTest {
+    private static final Logger log = LoggerFactory.getLogger(KubernetesPluginTest.class);
+
+    private static TServiceTemplate tServiceTemplate;
+    private static KubernetesPlugin plugin;
+    @Autowired
+    private CsarParseService csarParser;
+    @Autowired
+    private Preferences preferences;
+    @Autowired
+    private TestTransformationContext testTransformationContext;
+    @Autowired
+    private TestCsars testCsars;
+
+    @Before
+    public void setUp() {
+        plugin = new KubernetesPlugin();
+    }
+
+    @Test
+    public void validTServiceTemplate() throws InvalidDockerAppException, FileNotFoundException, InvalidCsarException {
+        Csar csar = testCsars.getCsar(CSAR_YAML_VALID_DOCKER_SIMPLETASK);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test(expected = InvalidDockerAppException.class)
+    public void dockerAppMissing() throws FileNotFoundException, InvalidCsarException, InvalidDockerAppException {
+        Csar csar = testCsars.getCsar(TestCsars.CSAR_YAML_INVALID_DOCKERAPP_MISSING);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test
+    public void transformTest() throws Exception {
+        File artifactRootPath = new File(preferences.getDataDir(), "test-csar/transformations/testplatform_one/");
+        TransformationContext transformationContext = testTransformationContext.getContext(CSAR_YAML_VALID_DOCKER_SIMPLETASK, TestPlugins.PLATFORM1);
+        assertNotNull(transformationContext);
+        plugin.transform(transformationContext);
+
+        checkFilesInTransformationRoot(artifactRootPath);
+        checkDockerFilesInTransformationRoot(artifactRootPath);
+    }
+
+    private void checkDockerFilesInTransformationRoot(File artifactRootPath) {
+        List<String> expectedDockerPathFiles = Lists.newArrayList("index.php", "mysql-credentials.php", "createdb.sql", "Dockerfile");
+        File transformationDockerFilesPath = new File(artifactRootPath, "/simple-task-app");
+        List<String> result = new LinkedList<>();
+        result.clear();
+        for (String s : transformationDockerFilesPath.list()) {
+            result.add(s);
+        }
+        assertTrue(expectedDockerPathFiles.containsAll(result));
+    }
+
+    private void checkFilesInTransformationRoot(File artifactRootPath) {
+        List<String> expectedTransformationRoot = Lists.newArrayList("simple-task-app", "simple-task-app_resource.yaml", "Readme.md", "test-csar-testplatform_one.log");
```

Comment:
I would completey change the philosophy of the test: I suggest you mock the PluginFileAccess and verify that the methods got called. E.g., verify that `accessMopluginFileAccess.copy("simple-task-app")` got called.